### DISCUSSION TITLE
An update on the path to the 1.0 Mod API release

### DISCUSSION THREAD

**Discussion by ConnorLinfoot** [HypixelDev/ModAPI#23] [Announcements]
Hello!

With the recent release of Mod API 0.4, we wanted to give an update on the current state of the Mod API, as well as its full 1.0 release:

* At this time we're in a good spot with the general structure of the base API and the protocol implementation, which are the most important parts for the full release.
* The current plan for version 1.0 of the Mod API is likely to include the available data that exists in this version.
  * While we are aware there is a lot of different data developers want we are starting with the basic information first.
  * Over the coming weeks and months, we will evaluate what data we can allow via the Mod API, and implement accordingly.
  * As a general reminder, the Mod API is not designed to replace the Public API, and thus a lot of data will not be something we wish to make available via the client/Mod API.
* Also, if you missed it, we have released the official Hypixel Mod API on Modrinth with support Fabric 1.20.5/1.20.6, and we have now shipped an implementation for Forge 1.8.9 too.
  * Both of these implementations are available for download on [Modrinth](https://modrinth.com/mod/hypixel-mod-api)
  * Our goal with these will be to keep both a Forge 1.8.9 build up to date, along with the latest version of Fabric.
  * If more versions or mod loaders are in high demand, we may consider adding support.
  * Mod developers should depend upon these mods in a way that allows multiple mods to take advantage of the API.
    * This can be achieved by requiring end-users to manually install the Hypixel Mod API, or at least with Fabric by deploying the Mod API with your jar as documented [here](https://support.modrinth.com/en/articles/8801191-modrinth-maven).
    * We also have plans to use the Wiki of this GitHub repo to include better examples and documentation of making use of the different implementations for both Fabric and Forge. We are also open to community contribution regarding this.
  * Lastly, if you are a Forge/Fabric developer and wish to provide improvements or fixes to their respective implementations, feel free to open a pull request in their respective repos that are linked in the [README](https://github.com/HypixelDev/ModAPI/blob/master/README.md).

Assuming no major issues pop up, and that the general consensus of the API remains. We will be aiming to release a 1.0 version in the coming weeks for developers to start using in their production mods. We plan to post any updates regarding the Mod API in this discussion section, so make sure to keep your eyes open for future announcements.

**Comment by SynicalMX** [comment]
> * Also, if you missed it, we have released the official Hypixel Mod API on Modrinth with support Fabric 1.20.5/1.20.6, and we have now shipped an implementation for Fore 1.8.9 too.

Typo in Forge.

**Comment by IndigoPolecat** [comment]
Hello,

I am developing a feature of my mod that will require semi-live knowledge of the player's current party status (with live being best if you feel like adding events for parties too :D) and I would much rather use the mod API instead of creating my own party detection via chat, but I am planning on releasing in a little over a month so I can't wait for 1.0 to be released before beginning development by the sounds of it. I wanted to know if the mod API was certain enough to develop on right now, with a potential release in ~5 weeks or if I should wait, or if you just can't say, I would appreciate any info!

Thank you!
